src/tests/functional/plugin/cpu/single_layer_tests/eye.cpp

```diff
@@ -0,0 +1,261 @@
+        rowNum = eyePar[0];
+        colNum = eyePar[1];
+        shift = eyePar[2];
+
+        init_input_shapes(inputShape);
+
+        selectedType = std::string("ref_I32");
+        function = createFunction();
+    }
+
+    std::shared_ptr<ngraph::Function> createFunction() {
+        auto inputParams = ngraph::builder::makeDynamicParams(ngraph::element::i32, inputDynamicShapes);
+        auto rowsPar = inputParams[0];
+        rowsPar->set_friendly_name("rows");
+        auto colsPar = inputParams[1];
+        colsPar->set_friendly_name("cols");
+        auto diagPar = inputParams[2];
+        diagPar->set_friendly_name("diagInd");
+        auto pars = ngraph::ParameterVector{rowsPar, colsPar, diagPar};
+        if (inputParams.size() == 4) {
+            auto batchShapePar = inputParams[3];
+            batchShapePar->set_friendly_name("batchShape");
+            pars.push_back(batchShapePar);
+            auto eyelikeBatchShape = std::make_shared<ngraph::op::v9::Eye>(rowsPar, colsPar, diagPar, batchShapePar, ngraph::element::i32);
+            eyelikeBatchShape->get_rt_info() = getCPUInfo();
+            return makeNgraphFunction(ngraph::element::i32, pars, eyelikeBatchShape, "Eye");
+        } else {
+            auto eyelikePure = std::make_shared<ngraph::op::v9::Eye>(rowsPar, colsPar, diagPar, ngraph::element::i32);
+            eyelikePure->get_rt_info() = getCPUInfo();
+            return makeNgraphFunction(ngraph::element::i32, pars, eyelikePure, "Eye");
+        }
+    }
+
+    void generate_inputs(const std::vector<ov::Shape>& targetInputStaticShapes) override {
+        inputs.clear();
+        const auto& funcInputs = function->inputs();
+        for (int i = 0; i < funcInputs.size(); ++i) {
+            const auto& funcInput = funcInputs[i];
+            ov::Tensor tensor;
+            if (i == 3) {  // batch shape
+                tensor = ov::Tensor(funcInput.get_element_type(), targetInputStaticShapes[i]);
+                int *batchShapePtr = tensor.data<int>();
+                for (size_t j = 0; j < targetInputStaticShapes[i].size(); j++) {
+                    batchShapePtr[j] = outBatchShape[j];
+                }
+            } else {
+                tensor = ov::test::utils::create_and_fill_tensor(funcInput.get_element_type(), targetInputStaticShapes[i], 1,
+                    (i == 0 ? rowNum : (i == 1 ? colNum : shift)));
+            }
+            inputs.insert({funcInput.get_node_shared_ptr(), tensor});
+        }
+    }
+};
+
+TEST_P(EyeLikeLayerCPUTest, CompareWithRefs) {
+    SKIP_IF_CURRENT_TEST_IS_DISABLED()
+    run();
+    CheckPluginRelatedResults(compiledModel, "Eye");
+}
+
+namespace {
+
+const std::vector<ElementType> netPrecisions = {
+    ElementType::f32, ElementType::bf16, ElementType::i32,
+    ElementType::i8, ElementType::u8
+};
+
+const std::vector<std::vector<int>> eyePars = {  // rows, cols, diag_shift
+    {3, 3, 0},
+    {3, 4, 1},
+    {3, 4, 10},
+    {4, 4, -2},
```

Comment:
Could you please add tests with number rows or columns are equal to zero?

Reply:
Added